### PR TITLE
[MRG+1] Update svg_tooltip.py

### PR DESCRIPTION
**Pull request by DaveL17**
Original example supported a maximum of three patches. Proposed change should support any number of patches.

### CODE REVIEW

File: examples/user_interfaces/svg_tooltip.py [hunk 1]
```diff
@@ -31,41 +31,22 @@
...
-              ec=(1., 1., 1.), lw=1, zorder=1))
-
-# Set id for the patches
-for i, t in enumerate(ax.patches):
-    t.set_gid('patch_%d' % i)
-
-# Set id for the annotations
-for i, t in enumerate(ax.texts):
-    t.set_gid('tooltip_%d' % i)
+rect1 = plt.Rectangle((10, -20), 10, 5, fc='blue')
```
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: Why change from the `Circle`?

File: examples/user_interfaces/svg_tooltip.py [hunk 2]
```diff
@@ -31,41 +31,22 @@
...
 
+shapes = [rect1, rect2]
+labels = ['This is a blue rectangle.', 'This is a green rectangle']
+
+for i, item in enumerate(shapes):
+    patch = ax.add_patch(item)
+    annotate = ax.annotate(labels[i], xy=item.get_xy(), xytext=(0, 0), 
+                           textcoords='offset points', color='w', ha='center', 
+                           fontsize=8, bbox=dict(boxstyle='round, pad=.5', fc=(.1, .1, .1, .92), 
+                           ec=(1., 1., 1.), lw=1, zorder=1))
```
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: This should be aligned with `dict(`.

File: examples/user_interfaces/svg_tooltip.py [hunk 3]
```diff
@@ -81,16 +62,14 @@
...
-for i, t in enumerate(ax.texts):
-    el = xmlid['tooltip_%d' % i]
-    el.set('visibility', 'hidden')
-
-# Assign onmouseover and onmouseout callbacks to patches.
-for i, t in enumerate(ax.patches):
-    el = xmlid['patch_%d' % i]
-    el.set('onmouseover', "ShowTooltip(this)")
-    el.set('onmouseout', "HideTooltip(this)")
+for i, y in enumerate(shapes):
```
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: `y` appears unused.

File: examples/user_interfaces/svg_tooltip.py [hunk 4]
```diff
@@ -31,41 +31,22 @@
...
-# Set id for the annotations
-for i, t in enumerate(ax.texts):
-    t.set_gid('tooltip_%d' % i)
+rect1 = plt.Rectangle((10, -20), 10, 5, fc='blue')
+rect2 = plt.Rectangle((-20, 15), 10, 5, fc='green')
 
+shapes = [rect1, rect2]
+labels = ['This is a blue rectangle.', 'This is a green rectangle']
+
+for i, item in enumerate(shapes):
```
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: `for i, (item, label) in enumerate(zip(shapes, labels)):`

**Comment by dstansby** [comment]
(fixes #8316)

File: examples/user_interfaces/svg_tooltip.py (hunk 2 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Agreed.

File: examples/user_interfaces/svg_tooltip.py (hunk 3 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Agreed.

File: examples/user_interfaces/svg_tooltip.py (hunk 1 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: I wasn't able to find a clean way to iterate the coordinates of a circle in the example for loop.  That's the only reason.

File: examples/user_interfaces/svg_tooltip.py (hunk 4 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Agreed.

File: examples/user_interfaces/svg_tooltip.py (hunk 1 above)
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: Interesting. Using a `CirclePolygon`, you can use the `.xy` property of both, but that requires an additional import from `matplotlib.patches` and it's not really a perfect circle.

With `Circle`, you can use the `.center` property, but then they're not exactly the same.

Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Right.  For the example, it didn't seem terribly pertinent that the shapes had to be different (I reckon other than to show that it works with different types of patches.)

File: examples/user_interfaces/svg_tooltip.py (hunk 3 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Hang on a sec.  It appears that `tooltip = xmlid['mytooltip_%d' % i]` throws a TypeError if y is not used.

Disagreed.

**Comment by DaveL17** [comment]
Just a note that the script works while patches are 10 or less. When the 11th patch is applied, the SVG will display the tooltip associated with the first patch. I suspect it's the `obj.id.slice(-1)` from the original example script.

File: examples/user_interfaces/svg_tooltip.py (hunk 3 above)
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: You're doing something wrong then; `y` should not be necessary.

File: examples/user_interfaces/svg_tooltip.py [hunk 5]
```diff
@@ -81,16 +65,14 @@
...
-for i, t in enumerate(ax.texts):
-    el = xmlid['tooltip_%d' % i]
-    el.set('visibility', 'hidden')
-
-# Assign onmouseover and onmouseout callbacks to patches.
-for i, t in enumerate(ax.patches):
-    el = xmlid['patch_%d' % i]
-    el.set('onmouseover', "ShowTooltip(this)")
-    el.set('onmouseout', "HideTooltip(this)")
+for i in enumerate(shapes):
```
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: `for i in range(len(shapes))`; don't need the `enumerate`.

File: examples/user_interfaces/svg_tooltip.py (hunk 3 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Addressed this concern in the latest commit.  I think all identified issues have been cleared.

File: examples/user_interfaces/svg_tooltip.py (hunk 5 above)
Comment (DaveL17) [inline examples/user_interfaces/svg_tooltip.py]: Agreed. Commit now references index instead.

File: examples/user_interfaces/svg_tooltip.py [hunk 6]
```diff
@@ -1,3 +1,6 @@
+#! /usr/bin/env python
```
Comment (QuLogic) [inline examples/user_interfaces/svg_tooltip.py]: It's not executable, so I don't think we need this (and I don't think it should be made executable either.)

File: examples/user_interfaces/svg_tooltip.py [hunk 7]
```diff
@@ -1,3 +1,6 @@
...
+# -*- coding: utf-8 -*-
+
 """
 SVG tooltip example
```
Comment (phobson) [inline examples/user_interfaces/svg_tooltip.py]: Could you add a row of `===` above this to make the title sphinx-gallery compliant?

**Comment by tacaswell** [comment]
Thanks @DaveL17 :tada:   Hopefully we will here from you again soon!

**Comment by tacaswell** [comment]
backported to v2.0.x as 41bd991db

**Pull request merged**